packages/store/src/-private/store-service.ts

```diff
@@ -238,7 +236,7 @@ class Store extends EmberObject {
     @private
   */
   constructor(createArgs?: Record<string, unknown>) {
-    super(createArgs);
+    super();
```

Comment:
I believe as long as we extend EmberObject we need to super the args to ensure services are available within `init` / after `super` should someone be extending the store and relying on that.

Reply:
this is the error, I wasn't sure what to do.

```
packages/store/src/-private/store-service.ts(239,11): error TS2345: Argument of type 'Record<string, unknown> | undefined' is not assignable to parameter of type 'Owner | undefined'.
  Type 'Record<string, unknown>' is missing the following properties from type 'Owner': register, lookup, factoryFor
```

`super(...arguments)` passes tsc but I don't know how correct that is

Reply:
Ember's actual API here is that owner is key'd onto the actual create args, owner itself is not the first argument. I think I remember @chriskrycho kinda fudged that for glimmer components for various reasons to keep it simpler (that or glimmer components take a different code path than everything else), but if the types are acting like this should be owner the types are wrong.

See implementation: https://github.com/emberjs/ember.js/blob/03e2e64166e39b2e3e3cd9661418830105344570/packages/%40ember/-internals/container/lib/container.ts#L527-L563

Reply:
If Ember's service types are wrong I suspect we should just @ts-expect-error that line, supering with the createArgs, and move on for now.